alphabase/smiles/peptide.py

```diff
@@ -0,0 +1,196 @@
+from typing import Dict, Optional
+
+from rdkit import Chem
+
+from alphabase.smiles.smiles import AminoAcidModifier
+
+
+class PeptideSmilesEncoder:
+    """
+    A class to encode peptide sequences into SMILES strings or RDKit molecule objects.
+    """
+
+    def __init__(self):
+        self.amino_acid_modifier = AminoAcidModifier()
+
+    def encode_peptide(
```

Comment:
Should we have a function `peptide_to_smiles` and `peptide_to_mol` to return the rdkit object if wanted?

Reply:
refactored to have both those methods